src/ixdat/readers/cinfdata_db.py

```diff
@@ -0,0 +1,364 @@
+
+        data_series_list = []
+        if self.verbose:
+            print("Column names in measurement: ")
+        for key in self.group_data.keys():
+            if self.group_meta[key]["type"] == 5:
+                column_name = self.group_meta[key]["mass_label"]
+                if self.verbose:
+                    print("Col name: ", column_name)
+                # unixtime = self.group_meta[key]["unixtime"]
+                # tstamp = float(unixtime)
+
+                tcol = self.group_data[key][:, 0]
+                vcol = self.group_data[key][:, 1]
+
+                tseries = TimeSeries(
+                    name=column_name + "-x",
+                    unit_name=get_column_unit(column_name + "-x") or "s",
+                    data=tcol,
+                    tstamp=self.tstamp,
+                )
+
+                vseries = ValueSeries(
+                    name=column_name,
+                    data=vcol,
+                    tseries=tseries,
+                    unit_name=get_column_unit(column_name + "-y"),
+                )
+                data_series_list.append(tseries)
+                data_series_list.append(vseries)
+            else:
+                pass
+
+        obj_as_dict = dict(
+            name=self.name,
+            sample_name=self.sample_name,
+            technique=self.technique,
+            reader=self,
+            series_list=data_series_list,
+            tstamp=self.tstamp,
+        )
+
+        if not obj_as_dict:
+            warnings.warn(
+                f"No mass spec data was found using '{self.token}' "
+                f" and group_column: '{self.grouping_column}'",
+                stacklevel=2,
+            )
+            return None
+
+        return obj_as_dict
+
+    def read_spectrums(self, **kwargs):
+        """Download spectrums from cinfdata_database"""
+        db = plugins.cinfdata.connect(
+            setup_name=self.setup_name, grouping_column=self.grouping_column
+        )
+
+        self.group_data = db.get_data_group(self.token)
+        self.group_meta = db.get_metadata_group(self.token)
+        self.meta = self.group_meta[list(self.group_meta.keys())[0]]
+
+        self.set_sample_name()
+        self.set_name()
+        self.set_tstamp()
+
+        spectrum_list = []
+        for i, key in enumerate(self.group_meta.keys()):
+            if self.group_meta[key]["type"] == 2:
+                self.x_name = "Binding energy / eV"
+                self.x_unit_name = "eV"
+                self.field_name = "Counts per second"
+                self.field_unit = "n/s"
+                self.technique = "XPS"
+                obj_as_dict = self.create_spectrum(key)  # group_data, group_meta, key)
+                obj_as_dict["name"] = self.group_meta[key]["name"]
+                spectrum_list.append(self.measurement_class.from_dict(obj_as_dict))
+
+            elif self.group_meta[key]["type"] == 4:
+                self.x_name = "Mass [AMU]"
```

Comment:
This could be misuse of self.

But this was the best way I could think of not copying all lines of code  in the method "create_spectrum()" which would be needed under each and every key["type"] 

that is of cause only two type now (type 2 and type 4). 

So my limited python knowlegde went with using self and an extra method to avoid copying existing lines of code. What is best to do?